src/Qir/microsoft-quantum-qir-runtime-sys/src/foundation.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use lazy_static::lazy_static;
+
+use libloading::Library;
+
+#[cfg(target_os = "linux")]
+const FOUNDATION_BYTES: &'static [u8] = include_bytes!(concat!(
+    env!("OUT_DIR"),
+    "/build/lib/QSharpFoundation/libMicrosoft.Quantum.Qir.QSharp.Foundation.so"
+));
+
+#[cfg(target_os = "macos")]
+const FOUNDATION_BYTES: &'static [u8] = include_bytes!(concat!(
+    env!("OUT_DIR"),
+    "/build/lib/QSharpFoundation/libMicrosoft.Quantum.Qir.QSharp.Foundation.dylib"
+));
+
+#[cfg(target_os = "windows")]
+const FOUNDATION_BYTES: &'static [u8] = include_bytes!(concat!(
+    env!("OUT_DIR"),
+    "/build/lib/QSharpFoundation/Microsoft.Quantum.Qir.QSharp.Foundation.dll"
+));
```

Comment:
Just FYI (I'm not saying whether it is good or bad):
I have seen a case where a single name for the dynamic library (written in Rust?) is used for all 3 platforms (in C#). See [here](https://github.com/microsoft/qsharp-runtime/blob/1334dc8cefb447e65feca66c463bcd77421bd5a2/src/Simulation/Simulators/OpenSystemsSimulator/NativeInterface.cs#L55), [here](https://github.com/microsoft/qsharp-runtime/blob/f6d9cf0292cf38bffff48d83fc1151207ed34429/build/pack.ps1#L32-L40), and [here](https://github.com/microsoft/qsharp-runtime/blob/5604fc6bdf31eb36e3e8ec577f416fc2426fcdee/src/Simulation/Common/Simulators.Dev.props#L35-L50). Written by Chris.

If that is not applicable to Rust then is it possible to write like this? (deduplicated)
```rust
const FOUNDATION_BYTES: &'static [u8] = include_bytes!(concat!(
    env!("OUT_DIR"),
#[cfg(target_os = "linux")]
    "/build/lib/QSharpFoundation/libMicrosoft.Quantum.Qir.QSharp.Foundation.so"
#[cfg(target_os = "macos")]
    "/build/lib/QSharpFoundation/libMicrosoft.Quantum.Qir.QSharp.Foundation.dylib"
#[cfg(target_os = "windows")]
    "/build/lib/QSharpFoundation/Microsoft.Quantum.Qir.QSharp.Foundation.dll"
));
```

Reply:
To clarify, I did that due to requirements in our CI and packaging pipeline for .NET projects; I wouldn't recommend following my approach in this case.